### PR TITLE
src/components/form: add FormCompanyChallenge form for creating company challenges

### PR DESCRIPTION
**Pull request by maceksimon**
Add component `FormCompanyChallenge` to be used for creating company challenges in Company Coordinator section.
The form will be used within a dialog, so it does not include title or submit buttons yet.

To allow entering start and end date, add `FormFieldDateRequired` with a popup date selector (Quasar component).
* Date input is tested within `FormFieldTestWrapper` component (same as other input fields)

* Add tests
* Add translations

Screenshot 2024-06-18 at 12 10 51

### CODE REVIEW

File: src/components/form/FormCompanyChallenge.vue [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+import { defineComponent, ref } from 'vue';
+
+// components
+import FormFieldDateRequired from '../form/FormFieldDateRequired.vue';
+import FormFieldTextRequired from '../global/FormFieldTextRequired.vue';
+
+// types
+import type { TransportType } from '../types/Route';
+type ChallengeType = 'regularity' | 'performance';
+type ChallengeParticipants = 'individuals' | 'teams' | 'branches';
```
Comment (tmszi) [inline src/components/form/FormCompanyChallenge.vue:33]: Instead of branches use subsidiaries, please.

File: src/i18n/en.toml [hunk 2]
```diff
@@ -120,13 +128,23 @@ labelJobTitle = "Your job title"
 labelJobTitleShort = "Job title"
 labelLastName = "Surname"
 labelNicknameOptional = "Nickname (optional)"
+labelParticipants = "Participant type"
+labelParticipantsIndividuals = "Individuals"
+labelParticipantsTeams = "Teams"
+labelParticipantsBranches = "Branches"
```
Comment (tmszi) [inline src/i18n/en.toml:134]: Subsidiaries

File: src/components/form/FormCompanyChallenge.vue (hunk 1 above)
Comment (maceksimon) [reply src/components/form/FormCompanyChallenge.vue:33]: corrected

File: src/components/form/FormCompanyChallenge.vue [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+        <!-- Teams -->
+        <q-radio
+          v-model="challengeParticipants"
+          val="teams"
+          data-cy="form-participants-teams"
+        >
+          <q-icon name="people" class="q-mr-xs text-grey-6" size="18px" />
+          {{ $t('form.labelParticipantsTeams') }}
+        </q-radio>
+        <!-- Branches -->
```
[the remark this answers is not in the document]
Comment (maceksimon) [reply src/components/form/FormCompanyChallenge.vue:116]: corrected

File: src/i18n/en.toml (hunk 2 above)
Comment (maceksimon) [reply src/i18n/en.toml:134]: corrected

File: src/components/form/FormCompanyChallenge.vue [hunk 4]
```diff
@@ -0,0 +1,270 @@
...
+        {{ $t('form.labelParticipants') }}
+      </legend>
+      <div class="q-gutter-sm">
+        <!-- Individuals -->
+        <q-radio
+          v-model="challengeParticipants"
+          val="individuals"
+          data-cy="form-participants-individuals"
+        >
+          <q-icon name="person" class="q-mr-xs text-grey-6" size="18px" />
```
Comment (tmszi) [inline src/components/form/FormCompanyChallenge.vue:104]: Icon size repeat > 1, could you set variable (constant) for this value, please?

Comment (maceksimon) [reply src/components/form/FormCompanyChallenge.vue]: Updated.

**Pull request merged**